packages/beacon-node/src/util/clock.ts

```diff
@@ -97,51 +105,63 @@ export class Clock extends EventEmitter implements IClock {
   get currentSlotWithGossipDisparity(): Slot {
     const currentSlot = this.currentSlot;
     const nextSlotTime = computeTimeAtSlot(this.config, currentSlot + 1, this.genesisTime) * 1000;
-    return nextSlotTime - Date.now() < MAXIMUM_GOSSIP_CLOCK_DISPARITY ? currentSlot + 1 : currentSlot;
+    return nextSlotTime - Date.now() < MAXIMUM_GOSSIP_CLOCK_DISPARITY_MS ? currentSlot + 1 : currentSlot;
   }
 
   get currentEpoch(): Epoch {
     return computeEpochAtSlot(this.currentSlot);
   }
 
-  /** Returns the slot if the internal clock were advanced by `toleranceSec`. */
-  slotWithFutureTolerance(toleranceSec: number): Slot {
-    // this is the same to getting slot at now + toleranceSec
-    return getCurrentSlot(this.config, this.genesisTime - toleranceSec);
+  /** Returns the slot if the internal clock were advanced by `toleranceMs`. */
+  slotWithFutureTolerance(toleranceMs: number): Slot {
+    const currentSlot = this.currentSlot;
+    const currentSlotStartTime = computeTimeAtSlot(this.config, currentSlot, this.genesisTime) * 1000;
+    const now = Date.now();
+    return toleranceMs > 0 && now > currentSlotStartTime && now - currentSlotStartTime <= toleranceMs
+      ? this.currentSlot - 1
+      : this.currentSlot;
   }
 
-  /** Returns the slot if the internal clock were reversed by `toleranceSec`. */
-  slotWithPastTolerance(toleranceSec: number): Slot {
-    // this is the same to getting slot at now - toleranceSec
-    return getCurrentSlot(this.config, this.genesisTime + toleranceSec);
+  /** Returns the slot if the internal clock were reversed by `toleranceMs`. */
+  slotWithPastTolerance(toleranceMs: number): Slot {
```

Comment:
same to the above comment, this function and the old version is not the same

Reply:
I added detail doc comment explaining the behavior of the function and the all the usage is also aligned with it.

Reply:
Why is the previous implementation not sufficient?

Reply:
Previous implementation was working with round `seconds` and truncating the any `ms`. I wanted to make the `clock` implementation consistently work with `ms` everywhere.